_adr/90/index.adoc

```diff
@@ -0,0 +1,136 @@
+
+7. There will be separate Letsencrypt's account per kas-fleet-manager environment. 
+For each of these acccounts, a request to overrides the various https://Letsencrypt.org/docs/rate-limits[Letsencrypt limits] will be done by following https://Letsencrypt.org/docs/rate-limits/#a-id-overrides-a-overrides[Letsencrypt limits overrides].
+There are two main limits that we should be aware of and are of interests: (1) Certificates per Registered Domain which has a default of 50 certificates per week (2) You can create a maximum of 300 New Orders per account per 3 hours.
+Each of these two limits needs to be bumped to a bigger value after capacity planning for certificate generation has been done.
+
+NOTE: There is an existing logic in the data plane that ensures that all sensitive parts in the ManagedKafka CR are not written in plain text. 
+Instead a secret is created and only references to the secrets will be added before the ManagedKafka is created in the data plane.
+
+### Threat model
+// Provide a link to the relevant threat model. 
+// You must either update an existing threat model(s) to cover the changes made by this ADR, or add a new threat model.
+
+Once the certificate are in the data plane in form of a Kubernetes secret, a privileged users can still see them. 
+If this is on a enterprise data plane cluster without good security standards/policy, then anyone with the private key can impersonate the server or eavesdrop on a connection with the server. 
+For non enterprise data plane clusters, this is mitigated by only allowing SREs to have the privilege / permissions to see these secrets.
+For enterprise data plane clusters, customers will be advised to do so as well.
+Morever, since there will be a different certificate per kafka instance, if one certificate is compromised, it won't affect others and it can be safely revoked. 
+The proposed solution also limits the possibility of an owner of a enterprise data plane cluster, eavesdropping on another data plane cluster that they do not own as the certificate keys will be different. 
+
+NOTE: If the shared certificate for an existing Kafka is to be compromised, then it opens up a securiry risk on all existing Kafkas. However, the chances of this security breach occurring is mitigated by first and foremost ensuring only SREs have privileged access to data plane clusters. And second, if a the security issue is detected then certificates can be easily revoked and new ones autocreated 
+
+## Alternatives Considered / Rejected
+
+1. Deploying https://www.redhat.com/sysadmin/cert-manager-operator-openshift[cert-manager operator] to the data plane. Rejected to avoid exposing domain registrar credentials and to avoid adding another component that consumes data plane resources and that needs to be monitored.
```

Comment:
I'm not an expert of this operator but does it rely on some external system or is it selfcontained/standalone?

I'm asking because one drawback of the current proposal is that Letsencrypt is going to be a SPOF for the service (at least during the creation of the instance) and we will probably need to consider in the future some sort of fallback to mitigate this risk

Reply:
I feel like the comment goes beyond the operator which I do not have much experience on but on how reliable Letsencrypt can be on how usage around it. So +1 to mitigate the risk of it becoming a SPOF - e.g by building retry mechanisms (some acme libraries have this) so that our SLOs are not impacted much.

Reply:
cert-manager can work with multiple CertificateIssuers, you can set these issuers up via CRs. Then in the Certificate Resource you can decide which issuer you want to use. So another way you can mitigate is to have a multiple ACME providers and if one if failing request the certificate from another provider